request-example.go

```diff
@@ -464,10 +464,19 @@ func (fs *root) Lstat(r *Request) (ListerAt, error) {
 	return listerat{file}, nil
 }
 
+// implements RealpathFileLister interface
+func (fs *root) Realpath(p string) string {
+	if fs.startDirectory == "" || fs.startDirectory == "/" {
+		return cleanPath(p)
+	}
+	return cleanPathWithBase(p, fs.startDirectory)
```

Comment:
It’s kind of weird that the only usage of this function occurs a file separate from its definition.

Usually, I prefer to see functions declared as close as possible to their usage, so that spacial locality is kept tight. But I do see `cleanPath` is used in a few different spots.